### DISCUSSION TITLE
How to output 24bit WAV files to I2SDAC

### DISCUSSION THREAD

**Discussion by kurohach** [micropython/micropython#11164] [General]
I am trying to build a high res music player using Pico and MicroPython.
Most high-resolution music sources are sampled at 24 bits, which must be output via I2S.

The I2S library supports 32 bits, and this 24-bit sound source can be shifted and played back 8 bits at a time.

To do this, I need to tuple each 24bit, but since arrays and structs do not support this, it does not work.

It is possible to convert the WAV file from 24-bit to 32-bit in advance, but I would like to avoid this because the file size will be too large.

How can I do this?

**Comment by Flipje1955** [comment]
I am currently playing with I2S, building on / using the work of Mike Teachman (see on Github) in particular combining I2S / WAV_files and using an SDCARD reader as - unless you work with very short clips - there is quickly insufficient space on the Pico's filesystem anyway. So, would an sdcard not remove your concern of the filesize ? 
By the way, I am using Audacity to reformat the WAV_files.  I usethe Pico's  I2S(0) and I2S(1) "in parallel", using asyncio. The Pico icw the sdcard is not able to source two  41000HZ / 32 bits coded WAF_files at the same time but for my case 16000Hz/16bits is ok and that works ok.
